### PR TITLE
Update logical replication security section

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [ ] Touched code is covered by tests
 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/admin/logical-replication.rst [hunk 1]
```diff
@@ -109,13 +110,14 @@ is not supported.
...
 --------
 
-To create a publication, a user must have the ``AL`` privilege on the cluster.
-To add tables to a publication, the user must have ``DQL``, ``DML``, ``DDL``
-privileges on the table. When user creates a publication that publishes all
-tables automatically, only those tables where user has ``DQL``, ``DML``,
-``DDL`` privileges will be published.
+To create, alter or drop a publication, a user must have the ``AL`` privilege
+on the cluster. To add tables to a publication, the user must have
+``DQL``, ``DML``, ``DDL`` privileges on the table. When user creates a
```
Comment (seut) [inline docs/admin/logical-replication.rst:115]:
```suggestion
``DQL``, ``DML``, ``DDL`` privileges on the table. When a user creates a
```

File: docs/admin/logical-replication.rst [hunk 2]
```diff
@@ -109,13 +110,14 @@ is not supported.
...
+publication that publishes all tables automatically, only those tables where
+user has ``DQL``, ``DML``, ``DDL`` privileges will be published.
```
Comment (seut) [inline docs/admin/logical-replication.rst:117]: Think we should add:
Only the owner (the user who created the publication) or a superuser is allowed to `ALTER` or `DROP` a publication.

And same for the subscription section below.

**Comment by seut** [comment]
While on this, could you also please fix the name of the referenced setting in the section? https://github.com/crate/crate/blob/master/docs/admin/logical-replication.rst?plain=1#L63

It currently is named `setting` while the referenced setting should be rendered.
Thanks!

File: docs/admin/logical-replication.rst [hunk 3]
```diff
@@ -109,13 +110,17 @@ is not supported.
...
+on the cluster. Only the owner (the user who created the publication) or a
+superuser is allowed to ALTER or DROP a publication.
+To add tables to a publication, the user must have
+``DQL``, ``DML``, ``DDL`` privileges on the table. When a user creates a
+publication that publishes all tables automatically, only those tables where
+user has ``DQL``, ``DML``, ``DDL`` privileges will be published.
+
+To create, alter or drop a subscription, a user must have the ``AL`` privilege
+on the cluster. Only the owner (the user who created the subscription) or a
+superuser is allowed to ALTER or DROP a publication.
```
Comment (seut) [inline docs/admin/logical-replication.rst:123]:
```suggestion
superuser is allowed to ALTER or DROP a subscription.
```

**Comment by BaurzhanSakhariev** [comment]
noticed not related to docs (but related to LR) failure https://jenkins.crate.io/job/CrateDB/job/crate_on_pr/616/testReport/junit/io.crate.integrationtests/MetadataTrackerITest/test_deleted_partition_is_replicated/

File: docs/admin/logical-replication.rst [hunk 4]
```diff
@@ -56,12 +56,13 @@ Every publication can have multiple subscribers.
...
+:ref:`ALTER PUBLICATION <sql-alter-publication>`
 
 .. CAUTION::
 
     Publishing cluster must have
-    :ref:`setting <sql-create-table-soft-deletes-enabled>`
+    :ref:`setting soft_deletes.enabled <sql-create-table-soft-deletes-enabled>`
     set to ``true`` so that subscribing cluster can catch up all changes made
     during replication pause caused by network issues or explicitly done by a user.
     Also, :ref:`setting <sql-create-table-soft-deletes-retention-lease-period>`
```
Comment (seut) [inline docs/admin/logical-replication.rst:68]:
```suggestion
    Also, :ref:`sql-create-table-soft-deletes-retention-lease-period`
```

File: docs/admin/logical-replication.rst [hunk 5]
```diff
@@ -56,12 +56,13 @@ Every publication can have multiple subscribers.
...
 altered or dropped using corresponding commands. The individual tables can be
-added and removed dynamically using ALTER PUBLICATION.
+added and removed dynamically using
+:ref:`ALTER PUBLICATION <sql-alter-publication>`
 
 .. CAUTION::
 
     Publishing cluster must have
-    :ref:`setting <sql-create-table-soft-deletes-enabled>`
+    :ref:`setting soft_deletes.enabled <sql-create-table-soft-deletes-enabled>`
```
Comment (seut) [inline docs/admin/logical-replication.rst:65]:
```suggestion
    :ref:`sql-create-table-soft-deletes-enabled` setting
```

**Review by seut — COMMENTED** [review]
fyi: references must not have a label if the original name is fine. labels inside references are useful to give the link a different name.
Sorry, but can you also make ``replication.logical.reads_poll_duration`` a reference by
:ref:`replication.logical.reads_poll_duration`?

**Comment by BaurzhanSakhariev** [comment]
> fyi: references must not have a label if the original name is fine. labels inside references are useful to give the link a different name. Sorry, but can you also make `replication.logical.reads_poll_duration` a reference by :ref:`replication.logical.reads_poll_duration`?

thx for the trick, removed names in SEE ALSO section as well in https://github.com/crate/crate/pull/12265/commits/a4daa56316a50ffc8cf1b235459142067ab02649

**Review by seut — APPROVED** [review]
👍

**Comment by seut** [comment]
some reference seem to be wrong a sphinx is failing...

**Comment by BaurzhanSakhariev** [comment]
> some reference seem to be wrong a sphinx is failing...

yep, cannot figure it out - for example, file contains 2 
```
:ref:`replication.logical.reads_poll_duration`
```
entries, one is rendered properly and one is not. Seems like has something to do with one link being inside SEE ALSO - trying currently different fixes

**Comment by seut** [comment]
@BaurzhanSakhariev seems like labels containing dots cannot be referenced without a name, so
```
:ref:`replication.logical.reads_poll_duration`
```

fails while:
```:ref:`replication.logical.reads_poll_duration<replication.logical.reads_poll_duration>```
succeeds.

Either use this, or change the label, eg. replace dots with _

**Comment by BaurzhanSakhariev** [comment]
retest this please

**Pull request merged**